dill/_dill.py

```diff
@@ -1049,6 +1026,56 @@ def _locate_function(obj, session=False):
     found = _import_module(obj.__module__ + '.' + obj.__name__, safe=True)
     return found is obj
 
+
+def _save_with_postproc(pickler, reduction, is_pickler_dill=None, obj=Getattr.NO_DEFAULT, postproc_list=None):
+    if obj is Getattr.NO_DEFAULT:
+        obj = Reduce(reduction) # pragma: no cover
+
+    if is_pickler_dill is None:
+        is_pickler_dill = is_dill(pickler, child=True)
+    if is_pickler_dill:
+        # assert id(obj) not in pickler._postproc, str(obj) + ' already pushed on stack!'
+        # if not hasattr(pickler, 'x'): pickler.x = 0
+        # print(pickler.x*' ', 'push', obj, id(obj), pickler._recurse)
+        # pickler.x += 1
+        if postproc_list is None:
+            postproc_list = []
+
+        # Recursive object not supported. Default to a global instead.
+        if id(obj) in pickler._postproc:
+            name = '(%s.%s) ' % (obj.__module__, getattr(obj, '__qualname__', obj.__name__)) if hasattr(obj, '__module__') else ''
+            warnings.warn('%s %shas strange recursive properties. Cannot pickle.' % (obj, name, PicklingWarning))
```

Comment:
I think throwing a warning is a good idea.  In keeping with convention, the warning should be something like: `warnings.warn("Can't pickle %r: %striggers a recursion error." % (obj, name, PicklingWarning))`.

Reply:
@anivegesana: That's better. I will very likely edit the warning to remove "perfectly" and "would" -- you can do it now, or I'll do it later.